contracts/Zauction.sol

```diff
@@ -12,11 +12,15 @@ contract ZAuction is Initializable, OwnableUpgradeable {
   using ECDSA for bytes32;
 
   IERC20 public token;
+  // IERC721 public nft;
   IRegistrar public registrar;
 
   // Original zAuction contract address for backward compatibility
   address legacyZAuction;
+
   mapping(address => mapping(uint256 => bool)) public consumed;
+  mapping(uint256 => uint256) public rootDomainId;
```

Comment:
Nit: Rename to `topLevelDomainIdCache` since this is a "cache" that gets used to reduce gas costs in future transactions.